common/processes/platform/tiny_process_library/tiny_process_library.cpp

```diff
@@ -77,23 +215,89 @@ ExpectedLineData Process::GenerateLineData() {
 			CollectLineData(trailing_line, ret, bytes, len);
 		});
 
-	auto id = proc_->get_id();
+	if (proc_->get_id() == -1) {
+		proc_.reset();
+		return expected::unexpected(MakeError(
+			ProcessesErrorCode::SpawnError,
+			"Failed to spawn '" + (args_.size() >= 1 ? args_[0] : "<null>") + "'"));
+	}
 
-	// waits for the process to finish
-	// TODO: log exit status != 0? Or error?
-	Wait();
+	SetupAsyncWait();
+
+	auto status = Wait(timeout);
+	if (!status) {
+		return expected::unexpected(
+			MakeError(NonZeroExitStatusError, "Collecting line data failed"));
+	}
 
 	if (trailing_line != "") {
 		ret.push_back(trailing_line);
 	}
 
-	if (id == -1) {
-		return expected::unexpected(MakeError(
-			ProcessesErrorCode::SpawnError,
-			"Failed to spawn '" + (args_.size() >= 1 ? args_[0] : "<null>") + "'"));
+	return ExpectedLineData(ret);
+}
+
+io::ExpectedAsyncReaderPtr Process::GetProcessReader(events::EventLoop &loop, int &pipe_ref) {
+	if (proc_) {
+		return expected::unexpected(
+			MakeError(ProcessAlreadyStartedError, "Cannot get process output"));
```

Comment:
Q: Why can't I get output from an already started process?

Reply:
We can make it possible, but what's the use case? I can't think of a case where starting a process output stream mid-way isn't a mistake.

Reply:
Yes, just wondering if there was any other justification :)